_posts/2017-07-05-ddd-modeling.md

```diff
@@ -0,0 +1,64 @@
+
+
+
+# The blocker scenario
+
+One of my favorite discussions centred around a common pattern developers could find within a given context, and how DDD ideas can help to think about it.
+In this scenario, we have a "big ball of mud" in which two bounded contexts collided. In one hand we have a "Draft" context, where the domain is the plan or negotiation process that the business wants to run. On the other hand, we have the "Execution" context which modelled the life cycle of an accepted and agreed plan. In other words, the "Draft" could be the negotiation of a cart in an e-commerce site and the "Execution" could be the life cycle of an accepted and paid cart.
+
+In this scenario, we have a piece of code that sticks together both contexts. Identifying that piece of code and working on separating the contexts gave an instant payoff. We were working within the core domain and were aiming to reduce the complexity of it, essentially reducing the cost of working within each of the contexts.
+
+
+![first image](/assets/img/custom/blog/2017-07-05-ddd-modeling-post/big-ball-of-mud.JPG)
+
+<img src="(/assets/img/custom/blog/2017-07-05-ddd-modeling-post/big-ball-of-mud.JPG" alt="Specification By Example" class="img-responsive"/>
+
+Prior to this, we had people from both sub-domains focusing on the same "big ball of mud". Splitting the bounded context separated the people that were interested in each context. Therefore, the people interested in the cart will point to the "Cart bounded context" and people interested in the order will point to the "Order bounded context".
+
+![second image](/assets/img/custom/blog/2017-07-05-ddd-modeling-post/cart-order.JPG)
+
+<img src="(/assets/img/custom/blog/2017-07-05-ddd-modeling-post/cart-order.JPG" alt="Specification By Example" class="img-responsive"/>
+
+Both bounded contexts share plenty of vocabulary in common. Which leads to more complex design and composing names in order to make obvious that a name belongs to the cart or to the order.
+
+An example of composite names:
+
+- \# CartTotaAmount -> Draft
+- \# OrderTotalAmount -> Execution
+- \# CartCanBeShipped -> Draft
+- \# OrderShipped -> Execution
+- \# CartConfirmed -> Draft
+- \# OrderConfirmed -> Execution
+
+
+# Conclusion
+
+Event Storming is a useful technique to understand the big picture of the business and spot the main blocker. The right people should be in the room when the Event Storming Session is in the room. The role of the facilitator is crucial.
+
+
+Domain Driven Design is awesome. It is not necessary to work following DDD to see that the ideas have plenty of value even if the team or other teams are not following DDD. It helps to understand your relationship with other teams as well as how to look problems from a domain perspective.
```

Comment:
The second sentence looks a bit redundant. My suggestion is to have something like:

_Even if not following all the principles, using just a few DDD ideas can already bring great value to the team._

In the last sentence, replace
_... to look problems ..._
with
_... to look at problems ..._